Automattic/woocommerce-payments
# Add Giropay Payment Method

Fixes #1510 

#### Changes proposed in this Pull Request
This PR is branched from the [Sepa PR](https://github.com/Automattic/woocommerce-payments/pull/1537) and shouldn't be merged until that one is merged to avoid conflicts. This PR adds the Giropay payment method. It is currently hidden behind a feature flag controlled by the `_wcpay_feature_giropay` option. This MVP implementation focuses on the successful payment flow using the classic editor. There will be more PRs to follow focusing on handling failed payment flows, block based checkout, and general optimization of the code.

**Edit:** This branch requires the payments server to be running the branch [here](740-gh-Automattic/woocommerce-payments-server) until it is merged. 

Some highlights to review:
* Allows the passing of additional fields to the API for creating an intent
* Adds the `next_action` and `last_payment_error` fields to the Payment Intention model
* Handles the `redirect_to_url` next action for intents with `requires_action` status
* Uses `wp` hook to check for the redirect back from Stripe and complete the order or error back to the checkout page
    * This hook was used in the [Stripe Gateway plugin](https://github.com/woocommerce/woocommerce-gateway-stripe), I didn't see any other obvious hook that fires early enough, but am open to other suggestions. 

### Testing instructions
* Set store currency to euro
* Add `_wcpay_feature_giropay` to `wp_options` table with a value of 1
* Enable Giropay by going to WooCommerce > Settings > Payments > Manage WooCommerce Payments (/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments) and using the checkbox at the bottom of the page.
* Add item to cart and go to checkout
* Choose Giropay and click Place Order
* You should be redirected to a Stripe page where you can view the payment information and choose to authorize or fail the payment.
* Choose Authorize and you should be redirected back to the order received page on the site.
* Check the Order in admin and see a successful payment and the proper payment notes recorded.
* Repeat the order process and choose fail to authorize the payment and you should be returned to the checkout page with an error displayed.

### Notes / Known Issues
[Stripe documentation](https://stripe.com/docs/payments/giropay/accept-a-payment#handle-redirect) used as reference
[Transaction details page for SEPA does not load in wcpay transaction tab](https://github.com/Automattic/woocommerce-payments/issues/1472 )


-------------------
